src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String WORD_SPLIT_REGEX = "\\W+";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        File inputFile = new File(fromFileName);
+        String[] words = readFromFile(inputFile).split(WORD_SPLIT_REGEX);
+
+        int supplySum = 0;
+        int buySum = 0;
+
+        for (int i = 0; i < words.length; i += 2) {
+            if (words[i].equals(SUPPLY)) {
+                supplySum += Integer.parseInt(words[i + 1]);
+            } else if (words[i].equals(BUY)) {
+                buySum += Integer.parseInt(words[i + 1]);
+            }
+        }
+
+        String report = generateReport(supplySum, buySum);
+        File outputFile = new File(toFileName);
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(outputFile))) {
+            bufferedWriter.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to create a file: " + outputFile.getName(), e);
+        }
+    }
```

Comment:
Logic of method is still overwhelmed.
Please extract part to new method like writeToFile() and other parts move to existing generateReport() and readFromFile() so public method look like below:
```suggestion
        String fileContent = readFromFile(fromFileName);
        String report = generateReport(fileContent);
        writeToFile(report, toFileName);
    }
```